src/Features/CSharp/Portable/BraceCompletion/CurlyBraceCompletionService.cs

```diff
@@ -101,40 +105,30 @@ public override async Task<bool> AllowOverTypeAsync(BraceCompletionContext conte
-                var indentation = indentationService.GetIndentation(document.WithText(formattedText), desiredCaretLine.LineNumber, cancellationToken);
-
-                // Insert whitespace for the indentation.
-                var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
-                var indentedText = GetIndentedText(indentation, documentOptions, formattedText);
+            // Set the caret position to the properly indented column in the desired line.
+            var newDocument = document.WithText(formattedText);
+            var newDocumentText = await newDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var caretPosition = GetIndentedLinePosition(newDocument, newDocumentText, desiredCaretLine.LineNumber, cancellationToken);
 
-                // The caret should be placed at the end of the indented line.
-                var caretLocation = indentedText.Lines[desiredCaretLine.LineNumber].End;
-                var overallChanges = indentedText.GetTextChanges(originalDocumentText).ToImmutableArray();
-                return new BraceCompletionResult(overallChanges, caretLocation);
-            }
+            var overallChanges = formattedText.GetTextChanges(originalDocumentText).ToImmutableArray();
+            return new BraceCompletionResult(overallChanges, caretPosition);
 
             static TextLine GetLineBetweenCurlys(int closingPosition, SourceText text)
             {
                 var closingBraceLineNumber = text.Lines.GetLineFromPosition(closingPosition - 1).LineNumber;
                 return text.Lines[closingBraceLineNumber - 1];
             }
 
-            static SourceText GetIndentedText(IndentationResult indentation, DocumentOptionSet documentOptions, SourceText originalText)
+            static LinePosition GetIndentedLinePosition(Document document, SourceText sourceText, int lineNumber, CancellationToken cancellationToken)
             {
-                var indentText = indentation.Offset.CreateIndentationString(documentOptions.GetOption(FormattingOptions.UseTabs), documentOptions.GetOption(FormattingOptions.TabSize));
-                var indentTextChange = new TextChange(new TextSpan(indentation.BasePosition, 0), indentText);
-                return originalText.WithChanges(indentTextChange);
+                var indentationService = document.GetRequiredLanguageService<IIndentationService>();
+                var indentation = indentationService.GetIndentation(document, lineNumber, cancellationToken);
+
+                var baseLinePosition = sourceText.Lines.GetLinePosition(indentation.BasePosition);
+                var offsetOfBacePosition = baseLinePosition.Character;
+                var totalOffset = offsetOfBacePosition + indentation.Offset;
+                var indentedLinePosition = new LinePosition(lineNumber, totalOffset);
+                return indentedLinePosition;
```

Comment:
not related to this code, just a general question:

Do our tests somehow validate the indentation position of brace completion?

Reply:
They do, both the caret and text
http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Test.Utilities/AutomaticCompletion/AbstractAutomaticBraceCompletionTests.cs,86